fiftyone/factory/repos/delegated_operation.py

```diff
@@ -310,13 +320,17 @@ def update_run_state(
             update["$set"]["status"] = progress
             update["$set"]["status"]["updated_at"] = datetime.utcnow()
 
+        collection_filter = {"_id": _id}
+        if required_state is not None:
+            collection_filter["run_state"] = required_state
+
         doc = self._collection.find_one_and_update(
-            filter={"_id": _id},
+            filter=collection_filter,
             update=[update],
             return_document=pymongo.ReturnDocument.AFTER,
         )
 
-        return DelegatedOperationDocument().from_pymongo(doc)
+        return DelegatedOperationDocument().from_pymongo(doc) if doc is not None else None
```

Comment:
If the method is update run state then if it fails to update shouldn't it error?

Reply:
This behavior is consistent with the underlying `find_one_and_update`, which returns `None` if no records match the filter. I generally avoid exceptions when there's a reasonable alternative but could be convinced otherwise if you feel strongly.